source/common/common/posix/thread_impl.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <functional>
+
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Thread {
+
+class ThreadIdImplPosix : public ThreadId {
+public:
+  ThreadIdImplPosix(int32_t id);
+
+  std::string debugString() const override;
+
+  bool isCurrentThreadId() const override;
+
+private:
+  int32_t id_;
+};
+
+/**
+ * Wrapper for a pthread thread. We don't use std::thread because it eats exceptions and leads to
+ * unusable stack traces.
+ */
+class ThreadImplPosix : public Thread {
+public:
+  ThreadImplPosix(std::function<void()> thread_routine);
+
+  /**
+   * Join on thread exit.
```

Comment:
Yeah, no need to repeat comments that are in the superclass, just a redirect to that is fine for overrides.